cmd/notation/sign.go

```diff
@@ -1,78 +1,93 @@
 package main
 
 import (
+	"context"
 	"fmt"
+	"time"
 
 	"github.com/notaryproject/notation-go"
 	"github.com/notaryproject/notation-go/crypto/timestamp"
 	"github.com/notaryproject/notation/internal/cmd"
 	"github.com/notaryproject/notation/internal/osutil"
 	"github.com/notaryproject/notation/pkg/config"
 	"github.com/opencontainers/go-digest"
-	"github.com/urfave/cli/v2"
+	"github.com/spf13/cobra"
 )
 
-var signCommand = &cli.Command{
-	Name:      "sign",
-	Usage:     "Signs artifacts",
-	ArgsUsage: "<reference>",
-	Flags: []cli.Flag{
-		cmd.FlagKey,
-		cmd.FlagKeyFile,
-		cmd.FlagCertFile,
-		cmd.FlagTimestamp,
-		cmd.FlagExpiry,
-		cmd.FlagReference,
-		flagLocal,
-		flagOutput,
-		&cli.BoolFlag{
-			Name:  "push",
-			Usage: "push after successful signing",
-			Value: true,
+type signOpts struct {
+	cmd.SignerFlagOpts
+	RemoteFlagOpts
+	timestamp       string
+	expiry          time.Duration
+	originReference string
+	output          string
+	push            bool
+	pushReference   string
+	pluginConfig    string
+	reference       string
+}
+
+func signCommand(opts *signOpts) *cobra.Command {
+	if opts == nil {
+		opts = &signOpts{}
+	}
+	command := &cobra.Command{
+		Use:   "sign [reference]",
+		Short: "Signs artifacts",
+		Args:  cobra.ExactArgs(1),
```

Comment:
Same here. The original logic's error is "missing reference", which looks more informative.